mutation/mutation.cc

```diff
@@ -180,7 +180,7 @@ mutation mutation::sliced(const query::clustering_row_ranges& ranges) const {
 
 mutation mutation::compacted() const {
     auto m = *this;
-    m.partition().compact_for_compaction(*schema(), always_gc, m.decorated_key(), gc_clock::time_point::min(), tombstone_gc_state(nullptr));
+    m.partition().compact_for_compaction(*schema(), always_gc, m.decorated_key(), gc_clock::time_point::min(), {});
```

Comment:
It would be nicer to have a name for it, no_gc, instead of an {} that one has to dive deeper to understand.

Reply:
It is not really `no_gc`, timeout based modes also work with default constructed state, just like before. Basically repair-mode is the only one which doesn't work with default-constructed gc getter.
Maybe we should replace default constructor with a static factory method which creates an object which has more intuitive semantics -- no-gc at all.

Reply:
Added named consrtuctors and removed the default one in v4.